business_rules/operators.py

```diff
@@ -576,13 +579,37 @@ def empty(self, other_value: dict):
         self.value[f"result_{uuid4()}"] = results
         return True in results
 
+    @type_operator(FIELD_DATAFRAME)
+    def empty_within_except_last_row(self, other_value: dict):
+        target = self.replace_prefix(other_value.get("target"))
+        comparator = other_value.get("comparator")
+        # group all targets by comparator
+        grouped_target = self.value.groupby(comparator)[target]
+        # validate all targets except the last one
+        results = grouped_target.apply(lambda x: x[:-1]).apply(pd.isnull)
```

Comment:
Does this ensure that the last row is not null? Or just that every row before the last row is null? I believe we also want to make sure the last row is not null